crates/amalthea/src/socket/shell.rs

```diff
@@ -452,25 +446,7 @@ impl Shell {
                 let handler = self.shell_handler.lock().unwrap();
 
                 // Call the shell handler to open the comm
-                match block_on(handler.handle_comm_open(comm, comm_socket.clone())) {
-                    Ok(result) => result,
-                    Err(err) => {
-                        // If the shell handler returns an error, send it back.
-                        // This is a language evaluation error, so we can send
-                        // it back in that form.
-                        let errname = err.ename.clone();
-                        req.send_error::<CommWireMsg>(err, &self.socket)?;
-
-                        // Return an error to the caller indicating that the
-                        // comm could not be opened due to the invalid open
-                        // call.
-                        return Err(Error::InvalidCommMessage(
-                            req.content.target_name.clone(),
-                            data_str,
-                            errname,
-                        ));
```

Comment:
Did we do lose this subtle behavior?

i.e. did we lose the `CommWireMsg` being sent back? And did we lose the special `InvalidCommMessage`?

Reply:
yes but these are not conform to the protocol. There is no reply to a `comm_open` so the error is logged on our end instead. We should reply a `comm_close` though. Here is an example impl: https://github.com/ipython/ipykernel/blob/3089438d84447bf859556b89f528ef59332803c5/ipykernel/comm/manager.py#L30

Reply:
We now reply a `comm_close` and this is now covered by a test.